packages/app/src/cli/services/dev/create-extension.ts

```diff
@@ -35,6 +35,7 @@ export async function createExtension(
     title: name,
     config: JSON.stringify({}),
     context: null,
+    handle: name,
```

Comment:
It looks like the `name` that gets passed into `createExtension` comes from either `extension.localIdentifier` or `extension.handle`. 

I took a look at the constructor for that class (`ExtensionInstance`), and it looks like `localIdentifier` is set by `handle` as well. 

https://github.com/Shopify/cli/blob/61862fec847b4843b81eb00c1c304b24121038c7/packages/app/src/cli/models/extensions/extension-instance.ts#L120-L121C10

What's the exact difference between the two?

cc: @isaacroldan

Reply:
the `localIdentifier` is a legacy property that we use to (as the name says) locally identify an extension. We used to use the folder name, but since the introduction of unified config we migrated to use the handle as the local identifier because a folder can contain multiple extensions now.
So right now `localIdentifier` === `handle`
And since the handle is optional, if there isn't one, we `slugify` the name.

For this specific method i'd say to pass the `handle` instead of the name to be more explicit:

```ts
export async function createExtension(
  apiKey: string,
  graphQLType: string,
  handle: string,
  token: string,
): Promise<ExtensionRegistration> {
  const query = ExtensionCreateQuery
  const variables: ExtensionCreateVariables = {
    apiKey,
    type: graphQLType,
    title: handle,
    config: JSON.stringify({}),
    context: null,
    handle: handle,
```